runtime/src/bank.rs

```diff
@@ -3045,6 +3046,8 @@ impl Bank {
         let partition_width = (PREFIX_MAX - partition_count + 1) / partition_count + 1;
         let mut start_key_prefix = if start_index == 0 && end_index == 0 {
             0
+        } else if start_index + 1 == partition_count {
```

Comment:
Change like this would need to be gated, right? It could change rent behavior. Also, is it specifically related or necessary for the scan fix?

Reply:
ah sorry,  I should've mentioned that this PR is dependent on another PR: https://github.com/solana-labs/solana/pull/13569#issuecomment-727121225 going in first, where gating will be necessary.